app/main.py

```diff
@@ -1,3 +1,57 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: (int, float)) -> None:
+        if km >= 0:
+            self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[Distance, int, float]) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
```

Comment:
Code quality issue: You're performing the same operation twice in the '__add__' method, once for 'Distance' objects and once for 'int' or 'float'. You can improve your code by converting 'other' to 'km' at the beginning of the method, and then perform the addition operation only once.